src/platform/events/rewards.ts

```diff
@@ -0,0 +1,79 @@
+import { generateLazyValidator, JSONSchema, ValidateFunction } from '../../validation'
+import { BaseEvent, Events } from './base'
+
+export type RewardEventMetadata = {
+  title: string
+  description: string
+  tokenName: string
+  tokenImage: string
+  tokenRarity: string | null
+  tokenCategory: string | null
```

Comment:
**NIT:**

By declaring properties like `string | null`, you must provide a value every time you create an object of that type:

```
myObject = {
  title: 'value',
  description: 'value',
  tokenName: 'value',
  tokenImage: 'value',
  tokenRarity: null
}
```

However, if we declare the properties as optional with the ? operator, like this:

```suggestion
  tokenRarity?: null
  tokenCategory?: null
```

Then, we can avoid initializing these properties if we don’t need to:

```
myObject = {
  title: 'value',
  description: 'value',
  tokenName: 'value',
  tokenImage: 'value',
  tokenRarity: null
}
```